project-base/storefront/hooks/comparison/useComparison.tsx

```diff
@@ -9,58 +8,43 @@ import {
 import { getUserFriendlyErrors } from 'helpers/errors/friendlyErrorMessageParser';
 import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
 import { useCurrentUserData } from 'hooks/user/useCurrentUserData';
-import { Dispatch, SetStateAction, useEffect, useMemo, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { usePersistStore } from 'store/zustand/usePersistStore';
 
-export const useHandleCompare = (
-    productUuid: string,
-): {
-    isProductInComparison: boolean;
-    handleProductInComparison: () => void;
-    handleRemoveAllFromComparison: () => void;
-    isPopupCompareOpen: boolean;
-    setIsPopupCompareOpen: Dispatch<SetStateAction<boolean>>;
-    getComparisonProducts: () => ComparedProductFragmentApi[];
-} => {
+export const useComparison = () => {
     const t = useTypedTranslationFunction();
-    const [isPopupCompareOpen, setIsPopupCompareOpen] = useState(false);
     const { isUserLoggedIn } = useCurrentUserData();
-    const [isProductInComparison, setIsProductInComparison] = useState(false);
     const [, addProductToComparison] = useAddProductToComparisonMutationApi();
     const [, removeProductFromComparison] = useRemoveProductFromComparisonMutationApi();
     const [, cleanComparison] = useCleanComparisonMutationApi();
-    const productsComparisonUuid = usePersistStore((store) => store.productsComparisonUuid);
+    const comparisonUuid = usePersistStore((store) => store.productsComparisonUuid);
     const updateUserState = usePersistStore((store) => store.updateUserState);
-    const [result] = useComparisonQueryApi({
-        variables: { comparisonUuid: productsComparisonUuid },
+    const [isPopupCompareOpen, setIsPopupCompareOpen] = useState(false);
+    const [pause, setPause] = useState(true);
+
+    const [{ data, fetching }] = useComparisonQueryApi({
```

Comment:
Just a nitpick: 

```suggestion
    const [{ data: comparisonData, fetching }] = useComparisonQueryApi({
```

I just dislike the generic names, such as 'result' or 'data'. If you disagree, you can just close this thread.

Reply:
No problem. Refactored.